docs/blog/2025-09-16-creating-an-nx-plugin-for-biome.md

```diff
@@ -0,0 +1,448 @@
+---
+title: 'Build Your Own Nx Plugin: Integrating Biome in 20 Minutes'
+slug: build-nx-plugin-biome-integration
+authors: ['Philip Fulcher']
+tags: ['nx']
+cover_image: /blog/images/2025-09-16/header.avif
+description: 'Learn how to create your own Nx plugin to integrate Biome, the fast Rust-based linter and formatter. Discover why writing your own plugin is often better than waiting for official support.'
+youtubeUrl: https://www.youtube.com/watch?v=GyDYN1AdfbQ
+---
+
+One question we get all the time is: "When are you going to support my tool of choice?" Whether it's a new linting tool, build tool, or framework, the truth is **we just can't support everything** with a team our size.
+
+That's exactly why we have an extensive plugin infrastructure with our devkit: so you can build that functionality yourself. And here's the thing: **you can absolutely do this**. We write our plugins using the exact same tools we provide to the community.
+
+Today, we're going to walk through integrating [Biome](https://biomejs.dev/), a fast linter and formatter written in Rust, into an Nx workspace. We'll show you how to get started with Biome, apply what you already know about the Node ecosystem, and finally write a plugin that makes Biome work seamlessly with Nx.
+
+{% toc /%}
+
+## Why Biome?
+
+Biome promises to be faster than ESLint and Prettier because it's written in Rust. And hey, who doesn't love [using Rust to make things fast](/blog/nx-15-8-rust-hasher-nx-console-for-intellij-deno-node-and-storybook#rustifying-the-nx-hasher)?
+
+We're working with a workspace that looks like this: a React application that depends on a few libraries. This workspace is small, but the process we're walking through will scale to any size workspace.
+
+![A workspace graph showing an application depending on three libraries.](/blog/images/2025-09-16/graph.avif)
+
+## Start with What You Know: Using Biome Without Nx
+
+Before thinking about Nx at all, let's see how far we can get knowing a little bit about Biome and a lot about the Node ecosystem.
+
+Looking at the [Biome getting started docs](https://biomejs.dev/guides/getting-started/), we need to:
+
+1. **Install the package**:
+
+```shell
+npm install --save-dev @biomejs/biome
+```
+
+2. **Create the root configuration** using their init command:
+
+```shell
+npx @biomejs/biome init
+```
+
+This creates a `biome.json` file at the root of our repo.
+
+3. **Run the linter**:
+
+```shell
+npx @biomejs/biome lint
+```
+
+Without thinking about Nx at all, we've successfully installed Biome and run a lint across our entire workspace. That's a good starting point, but in a monorepo, **we don't want to lint all files all the time**. We want to lint individual projects.
+
+## Using npm scripts
```

Comment:
some of the titles are not coherent in the casing that's being used (sentence casing vs title casing)